packages/client/utils/retroGroup/updateClonePosition.ts

```diff
@@ -25,22 +52,47 @@ const getTransition = (isClipped: boolean, timeRemaining: number) => {
 }
 
 export const getDroppingStyles = (
-  targetEl: HTMLDivElement,
+  element: HTMLDivElement,
   bbox: ClientRect,
   maxTop: number,
-  timeRemaining: number
+  timeRemaining: number,
+  targetId?: string | null,
+  isClose?: boolean,
+  lastStyle?: React.CSSProperties
 ) => {
   const spotlightEl = document.getElementById('spotlight')
-  const showAboveSpotlight = spotlightEl ? spotlightEl.contains(targetEl) : false
+  const isTargetInSpotlight = targetId
+    ? spotlightEl && document.querySelector(`div[${DragAttribute.DROPPABLE}='${targetId}']`)
+    : false
+  const isInSpotlight = spotlightEl && spotlightEl.contains(element)
+  const showAboveSpotlight = isInSpotlight || isTargetInSpotlight
   const {top, left} = bbox
-  const minTop = getMinTop(top, targetEl)
+  const minTop = getMinTop(top, element)
   const clippedTop = Math.min(Math.max(minTop, top), maxTop - bbox.height)
   const isClipped = clippedTop !== top
+
+  const fadeInAnimation = `${reflectionSpotlightFadeIn.toString()} 0.5s linear 0s forwards`
+  const fadeOutAnimation = `${reflectionSpotlightFadeOut.toString()} 0.5s linear 0s forwards`
+
+  const isCurrentlyBehindSpotlight =
+    lastStyle &&
+    (lastStyle.zIndex === ZIndex.REFLECTION_IN_FLIGHT || lastStyle.zIndex === undefined)
```

Comment:
It looks like we only need the `zIndex` from `lastStyle`. Rather than storing `lastStyle` in local state in the parent component, could we just store `lastZIndex` in local state or a ref?

Reply:
@nickoferrall As far as I see, we are unable to use local state here as it is just a plain function calculating value, not a component.